src/sqlodbc/opensearch_communication.cpp

```diff
@@ -548,26 +564,31 @@ bool OpenSearchCommunication::CheckSQLPluginAvailability() {
 }
 
 bool OpenSearchCommunication::EstablishConnection() {
-    // Generate HttpClient Connection class if it does not exist
+
     LogMsg(OPENSEARCH_ALL, "Attempting to establish DB connection.");
+
+    // Generate HttpClient Connection class if it does not exist
     if (!m_http_client) {
         InitializeConnection();
     }
 
-    // check if the endpoint is initialized
-    if (sql_endpoint.empty()) {
-        SetSqlEndpoint();
+    SetIsServerless();
+    SetSqlEndpoint();
+
+    if (sql_endpoint == SQL_ENDPOINT_ERROR) {
```

Comment:
Can we add a comment here clarifying how connecting to serverless affects these checks? It's hard to follow.

Reply:
Added some comments. Also update this `if` clause to also check `is_serverless` (even though this is unnecessary, it hopefully makes the dependencies a bit more clear). Let me know what you think!